nuxeo-runtime/nuxeo-runtime-stream/src/main/java/org/nuxeo/runtime/avro/AvroSchemaFactoryServiceImpl.java

```
@@ -0,0 +1,76 @@
+ * limitations under the License.
+ *
+ * Contributors:
+ *     pierre
+ */
+package org.nuxeo.runtime.avro;
+
+import java.lang.reflect.Constructor;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * @since 10.2
+ */
+public class AvroSchemaFactoryServiceImpl implements AvroSchemaFactoryService {
+
+    protected static final Log log = LogFactory.getLog(AvroSchemaFactoryServiceImpl.class);
+
+    protected final SortedSet<AvroReplacementDescriptor> replacements = new TreeSet<>();
+
+    protected Map<Class<?>, Class<AvroSchemaFactory<?>>> factories = new HashMap<>();
+
+    public AvroSchemaFactoryServiceImpl(Collection<AvroReplacementDescriptor> replacements) {
+        for (AvroReplacementDescriptor replacement : replacements) {
+            this.replacements.add(replacement);
+        }
+    }
+
+    public AvroSchemaFactoryServiceImpl() {
+        this(Collections.emptyList());
+    }
+
+    public void setFactories(Map<Class<?>, Class<AvroSchemaFactory<?>>> factories) {
+        this.factories.clear();
+        this.factories.putAll(factories);
+    }
+
+    @Override
+    public AvroSchemaFactoryContext createContext() {
+        AvroSchemaFactoryContext context = new AvroSchemaFactoryContext(replacements);
+        for (Entry<Class<?>, Class<AvroSchemaFactory<?>>> entry : factories.entrySet()) {
+            try {
+                Class<AvroSchemaFactory<?>> clazz = entry.getValue();
+                Constructor<AvroSchemaFactory<?>> constructor = clazz.getConstructor(AvroSchemaFactoryContext.class);
+                AvroSchemaFactory<?> factory = constructor.newInstance(context);
+                context.register(entry.getKey(), factory);
+            } catch (Exception e) {
+                log.error(e);
```

Please use a more specific exception than `Exception`, and rethrow instead of logging.